Public/Src/Cache/ContentStore/Library/Utils/ResourcePool.cs

```diff
@@ -181,10 +183,10 @@ public async Task<ResourceWrapper<TObject>> CreateAsync(TKey key)
                     // Start resource "GC" if the cache is full and it isn't already running
                     await EnsureCapacityAsync();
 
-                    returnWrapper = _resourceDict.GetOrAdd(key, k =>
-                    {
-                        return new ResourceWrapper<TObject>(() => _resourceFactory(k));
-                    });
+                    returnWrapper = _resourceDict.GetOrAdd(
+                        key,
+                        (k, resourceFactory) => { return new ResourceWrapper<TObject>(() => resourceFactory(k)); },
+                        _resourceFactory);

```

Comment:
Why pass in the factory instead of just calling it? To avoid the capture?

Reply:
Yep. To avoid closure allocation on every call.

---
In reply to: [330168861](https://github.com/microsoft/BuildXL/pull/968#discussion_r330168861) [](ancestors = 330168861)